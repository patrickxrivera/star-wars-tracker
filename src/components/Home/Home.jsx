import React from 'react';
import { Container, Title, Subscript } from './HomeStyles.jsx';

import CategoriesContainer from '../Categories/CategoriesContainer/CategoriesContainer.jsx';

const Home = () => (
  <Container>
    <Title>Select a category</Title>
    <Subscript>
      Detailed information on your favorite Star Wars characters.
    </Subscript>
    <CategoriesContainer />
  </Container>
);

export default Home;
