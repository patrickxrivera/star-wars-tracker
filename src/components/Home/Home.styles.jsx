import styled, { css } from 'styled-components';
import { fadeInUp } from '../../styles/animations.js';

export const Container = styled.div`
  margin: 80px auto 0;
  display: flex;
  flex-direction: column;
  padding-right: 3.8rem;
  opacity: 0;
  animation: 1100ms ${fadeInUp} 1500ms forwards;
`;

export const Title = styled.span`
  font-size: 3.5rem;
  text-align: center;
`;

export const Subscript = styled.span`
  text-align: center;
  margin-top: 3.5rem;
  font-size: 1.5rem;
`;

export const CategoriesContainer = styled.div`
  display: flex;
  flex: 1;
  justify-content: center;
  margin-top: 4.5rem;
`;

export const CategoryStyles = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: ${(props) => props.color};
  margin: 0 3rem;

  &:hover {
    opacity: 0.6;
    cursor: pointer;
  }
`;

export const CategoryName = styled.span`
  font-size: 1.25rem;
  margin-top: 0.6rem;
`;
