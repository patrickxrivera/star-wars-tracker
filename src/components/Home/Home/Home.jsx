import React, { Component } from 'react';
import styled from 'styled-components';
import icons from './HomeData.jsx';
import {
  Container,
  Title,
  Subscript,
  CategoriesContainer,
  CategoryStyles,
  CategoryName
} from './Home.styles.jsx';

const Category = styled.div`
  ${CategoryStyles};
`;

class Home extends Component {
  constructor(props) {
    super(props);

    [
      'handleClick',
      'getJSXForCategories',
      'getIconDataFrom',
      'createJSXFrom'
    ].forEach((propToBind) => {
      this[propToBind] = this[propToBind].bind(this);
    });
  }

  handleClick() {
    return null;
  }

  getJSXForCategories() {
    const categoriesJSX = Object.keys(icons).map((categoryName) => {
      const iconData = this.getIconDataFrom(icons, categoryName);
      const categoryJSX = this.createJSXFrom(iconData);
      return categoryJSX;
    });

    return categoriesJSX;
  }

  getIconDataFrom(icons, categoryName) {
    const [color, Icon] = [
      icons[categoryName].color,
      icons[categoryName].iconName
    ];
    return { color, Icon, categoryName };
  }

  createJSXFrom({ color, Icon, categoryName }) {
    return (
      <Category
        onClick={this.handleClick}
        key={categoryName}
        color={color}
        data-test={`category-${categoryName}`}
      >
        <Icon size={35} />
        <CategoryName>{categoryName}</CategoryName>
      </Category>
    );
  }

  render() {
    const categoriesJSX = this.getJSXForCategories();

    return (
      <Container>
        <Title>Select a category</Title>
        <Subscript>
          Detailed information on your favorite Star Wars characters.
        </Subscript>
        <CategoriesContainer>{categoriesJSX}</CategoriesContainer>
      </Container>
    );
  }
}

export { Container, CategoriesContainer, Category };
export default Home;
