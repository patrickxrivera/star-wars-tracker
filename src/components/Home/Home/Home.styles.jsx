import styled, { css } from 'styled-components';
import { fadeInUp } from '../../../styles/animations.js';

export const Container = styled.div`
  margin: 80px auto 0;
  display: flex;
  flex-direction: column;
  padding-right: 3.8rem;
  opacity: 0;
  animation: 1100ms ${fadeInUp} 1500ms forwards;
`;

export const Title = styled.span`
  font-size: 3.5rem;
  text-align: center;
`;

export const Subscript = styled.span`
  text-align: center;
  margin-top: 3.5rem;
  font-size: 1.5rem;
`;
