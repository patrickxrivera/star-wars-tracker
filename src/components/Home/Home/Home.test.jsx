import React from 'react';
import { mount } from 'enzyme';
import Home from './Home.jsx';

test('Home renders correctly', () => {
  const wrapper = mount(<Home />);
  expect(wrapper).toMatchSnapshot();
});
