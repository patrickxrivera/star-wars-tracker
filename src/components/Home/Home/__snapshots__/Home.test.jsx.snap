// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<Home>
  <styled.div>
    <div
      className="sc-bdVaJa hlMorZ"
    >
      <styled.span>
        <span
          className="sc-bwzfXH dvooCS"
        >
          Select a category
        </span>
      </styled.span>
      <styled.span>
        <span
          className="sc-htpNat igytiI"
        >
          Detailed information on your favorite Star Wars characters.
        </span>
      </styled.span>
      <styled.div>
        <div
          className="sc-bxivhb cjKwyO"
        >
          <styled.div
            color="#4f5ef5"
            data-test="category-People"
            key="People"
            onClick={[Function]}
          >
            <div
              className="sc-EHOje dLmORQ"
              color="#4f5ef5"
              data-test="category-People"
              onClick={[Function]}
            >
              <User
                color="currentColor"
                size={35}
              >
                <svg
                  fill="none"
                  height={35}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={35}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
                  />
                  <circle
                    cx="12"
                    cy="7"
                    r="4"
                  />
                </svg>
              </User>
              <styled.span>
                <span
                  className="sc-ifAKCX djuxRU"
                >
                  People
                </span>
              </styled.span>
            </div>
          </styled.div>
          <styled.div
            color="#f44c67"
            data-test="category-Planets"
            key="Planets"
            onClick={[Function]}
          >
            <div
              className="sc-EHOje dmjSaJ"
              color="#f44c67"
              data-test="category-Planets"
              onClick={[Function]}
            >
              <Chrome
                color="currentColor"
                size={35}
              >
                <svg
                  fill="none"
                  height={35}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={35}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="4"
                  />
                  <line
                    x1="21.17"
                    x2="12"
                    y1="8"
                    y2="8"
                  />
                  <line
                    x1="3.95"
                    x2="8.54"
                    y1="6.06"
                    y2="14"
                  />
                  <line
                    x1="10.88"
                    x2="15.46"
                    y1="21.94"
                    y2="14"
                  />
                </svg>
              </Chrome>
              <styled.span>
                <span
                  className="sc-ifAKCX djuxRU"
                >
                  Planets
                </span>
              </styled.span>
            </div>
          </styled.div>
          <styled.div
            color="#2fca74"
            data-test="category-Vehicles"
            key="Vehicles"
            onClick={[Function]}
          >
            <div
              className="sc-EHOje fFnlGC"
              color="#2fca74"
              data-test="category-Vehicles"
              onClick={[Function]}
            >
              <Send
                color="currentColor"
                size={35}
              >
                <svg
                  fill="none"
                  height={35}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={35}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="22"
                    x2="11"
                    y1="2"
                    y2="13"
                  />
                  <polygon
                    points="22 2 15 22 11 13 2 9 22 2"
                  />
                </svg>
              </Send>
              <styled.span>
                <span
                  className="sc-ifAKCX djuxRU"
                >
                  Vehicles
                </span>
              </styled.span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Home>
`;
